src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -0,0 +1,138 @@
+from pyspiffe.workloadapi.workload_api_client import WorkloadApiClient
+from pyspiffe.internal.defaults import Defaults
+
+class DefaultWorkloadApiClient(WorkloadApiClient):
+    """
+    Default implementation for a SPIFFE Workload API Client.
+    """
+
+    # TODO: Not sure we need all the arguments, see newClient()
+    def __init__(self, spiffe_socket_path=None, exponential_backoff_policy=None,
+                 executor_service=None):
+
+        if spiffe_socket_path == None:
+            self.spiffe_socket_path=Defaults.spiffe_socket_path
+        else:
+            self.spiffe_socket_path=spiffe_socket_path
+
+        # TODO defaults for these, or delete them entirely
+        self.exponential_backoff_policy=exponential_backoff_policy
+        self.executor_service=executor_service
+
+
+    # TODO: Is this pythonic? Any benefit of this being a factory object vs
+    # just using the constructor?
+    @classmethod
+    def new_client(cls, spiffe_socket_path=None, exponential_backoff_policy=None,
+                  executor_service=None):
+        """
+        Create a new Workload API Client.
+
+        Parameters:
+
+        spiffeSocketPath (String) [optional]: Path to Workload API UDS.
+        exponentialBackoffPolicy (?? Type): Custom backoff policy instance. Do we need?
+        executorService (?? Type): Custom gRPC executor. Do we need?
+
+        Returns:
+
+        DefaultWorkloadApiClient: New Workload API Client object.
+        """
+
+        return cls(spiffe_socket_path=spiffe_socket_path,
+                   exponential_backoff_policy=exponential_backoff_policy,
+                   executor_service=executor_service)
+
+
+    def fetch_x509_context(self):
+        """
+        Fetches an X.509 context on a one-shot blocking call.
+
+        Fetch an instance of an X509Context containing the X.509
+        materials fetched from the Workload API,
+
+        Returns:
+
+        X509Context: The X.509 context.
+        """
+
+        pass
+
+
+    def watch_x509_context(self, watcher):
+        """
+        Watches for X.509 context updates.
+
+        A new Stream to the Workload API is opened for each call to
+        this method, so that the client starts getting updates
+        immediately after the Stream is ready and doesn't have to wait
+        until the Workload API dispatches the next update based on the
+        SVIDs TTL.
+
+        Parameters:
+
+        watcher (?? type): Watcher callback object.
+        """
+
+        pass
+
+
+    def fetch_jwt_svid(self, audiences, subject=None):
```

Comment:
Let's include type hints  to follow [PEP 484](https://www.python.org/dev/peps/pep-0484/)
For example:
```
def greeting(name: str) -> str:
    return 'Hello ' + name
```

Reply:
Thanks, I'll look into this.

Reply:
Should be resolved, but please verify as this is something I haven't used in Python before.

Reply:
I've just added a comment indicating that we should also include type information for method parameters. 
Thanks for the patience Scott.

Reply:
Whoops, I missed the hint on parameters,  I only did the return type. I'll fix this, thanks!